neuralop/layers/spectral_convolution_laplace.py

```diff
@@ -0,0 +1,979 @@
+        Returns
+        -------
+        tuple
+            Tuple of (transient part, steady-state part)
+        """
+        # Initialize output tensor for frequency response
+        Hw = torch.zeros(
+            weights_residue.shape[0],
+            weights_residue.shape[0],
+            weights_residue.shape[2],
+            weights_residue.shape[3],
+            weights_residue.shape[4],
+            lambda1.shape[0],
+            lambda2.shape[0],
+            lambda2.shape[3],
+            device=alpha.device,
+            dtype=torch.cfloat,
+        )
+
+        # Calculate pole-residue representation in the 3D Laplace domain
+        # by combining poles from all three dimensions
+        term1 = torch.div(
+            1,
+            torch.einsum(
+                "pbix,qbik,rbio->pqrbixko",
+                torch.sub(lambda1, weights_pole1),
+                torch.sub(lambda2, weights_pole2),
+                torch.sub(lambda3, weights_pole3),
+            ),
+        )
+        Hw = torch.einsum("bixko,pqrbixko->pqrbixko", weights_residue, term1)
+
+        # Compute both transient and steady-state parts
+        # Note: For 3D PDEs, the steady-state part uses negative Hw
+        output_residue1 = torch.einsum("bioxs,oxsikpqr->bkoxs", alpha, Hw)
+        output_residue2 = torch.einsum("bioxs,oxsikpqr->bkpqr", alpha, -Hw)
+        return output_residue1, output_residue2
+
+    def forward(self, x: torch.Tensor, output_shape: Optional[Tuple[int]] = None):
+        """Forward pass for 3D Laplace spectral convolution.
+
+        Parameters
+        ----------
+        x : torch.Tensor
+            Input tensor of shape (batch_size, channels, D, H, W)
+        output_shape : tuple, optional
+            Desired output spatial dimensions, by default None
+
+        Returns
+        -------
+        torch.Tensor
+            Output tensor of shape (batch_size, out_channels, D, H, W) or output_shape
+        """
+        batchsize, channels, *mode_sizes = x.shape
+
+        # Setup computation parameters
+        modes1, modes2, modes3 = self.n_modes
+        D, H, W = x.shape[-3], x.shape[-2], x.shape[-1]
+        modes1, modes2, modes3 = min(modes1, D), min(modes2, H), min(modes3, W)
+        self.linspace_steps = x.shape[2:]
+
+        # Compute the grid and spacing for the domain
+        dt_list, shape = _compute_dt(
+            shape=self.linspace_steps,
+            start_points=self.linspace_startpoints,
+            end_points=self.linspace_endpoints,
+        )
+        tz = shape[0].to(x.device)
+        tx = shape[1].to(x.device)
+        ty = shape[2].to(x.device)
+        dtz = dt_list[0]  # this can be time dimension, instead of Z dimension
+        dtx = dt_list[1]
+        dty = dt_list[2]
+
+        # Transform to frequency domain and compute frequency grids
+        alpha = torch.fft.fftn(x, dim=[-3, -2, -1])
+        omega1 = (
+            torch.fft.fftfreq(tz.shape[0], dtz, device=alpha.device) * 2 * np.pi * 1j
```

Comment:
These fftfreq operations could be chained using iterables/loops to abstract away dimensions in this part